src/components/select-field/BaseSelectField.js

```diff
@@ -68,6 +68,8 @@ type Props = {
     options: Array<SelectOptionProp>,
     /** The select button text shown when no options are selected. */
     placeholder?: string | React.Element<any>,
+    /** Boolean to determine whether the dropdown should be in fixed position */
+    positionFixed?: boolean,
```

Comment:
We have had this crop up a few times before; when a component wraps some library component, how should we handle that libraries built-in options?

Here, we have a one-off case handled. `positionFixed` is added directly to the signature of BaseSelectField, but it is actually determined by `react-popper`. Popper props are mixed with this component's props. 

IMO, it is best to provide an object prop for the library's options. that way, there are fewer future changes to the component, and it is easier to refactor and type. for instance, popper may have a type for the options, which we could use here.

[`Popper.Defaults`](https://popper.js.org/docs/v1/#popperdefaults--codeobjectcode) is defined here.

Reply:
Looked at `Popper.Defaults` and by passing in these static properties, we are passing these into the Popper.Defaults object. The Popper component accepts the following properties defined here: https://www.npmjs.com/package/react-popper/v/1.3.5

Added it to BaseSelectField as an optional prop to leave the responsibility of overriding the default to whoever is implementing it. in this case, EUA would be the one implementing the MultiSelect component and can pass down whether they want to override `react-popper`'s default positioning.

This is also following the current design of `placement` for `dropdownPlacement` and `modifiers` for `dropdownModifiers`.

Reply:
What I mean is that all the other props for BaseSelectField are specific to BaseSelectField. `dropdownPlacement` and `dropdownModifiers` are derived from props passed in that are not Popper-specific. By adding one of Popper's expected parameters directly to this component's props, we do to things:

- We tie BaseSelectField's implementation to the external library
- We establish the pattern of directly copying all future defaults into BaseSelectField's API

An object would make sense if `fixedPosition` were grouped in with other things, but it appears to be at the same level as the other defaults, as props on `<Popper/>`.

To match what is being done in BaseSelectField, we might add an `isPositionFixed` optional prop, and do a similar check on ln592 that we do for the other popper values, to create the popper-specific prop for `<PopperComponent/>`.

The other benefit is that we can avoid some of the confusion if we upgrade to popper 2.x, since they change the parameters around a lot.